grammar.js

```diff
@@ -448,12 +448,9 @@ const rules = {
   function_type_specifier: $ =>
     seq(
       rep($._type_modifier),
-      '(',
-      'function',
-      '(',
+      /\(\s*function\s*\(/,
       com.opt(opt($.inout_modifier), $._type, opt($.variadic_modifier), ','),
-      ')',
-      ':',
+      /\)\s*:/,
```

Comment:
Okay I have one more for you. You can actually use comments between the function type specifier tokens. But I'm thinking we can merge this PR and we can create an issue to solve this separately. We should support whatever Hack supports, but any more than spaces between these tokens seems unlikely in real code.

I would verify that we don't increase the errors in the ocaml-tree-sitter tests (since they test a lot more code so they might hit the edge cases).

Reply:
I think we can alleviate the impact of this by being less strict with the changes. For this fix to work, I think we only need to merge two of the rule items. Ex. the rule below works and allows comments except between function keyword and first parenthesis:
```
function_type_specifier: $ =>
  seq(
    rep($._type_modifier),
    '(',
    /function\s*\(/,
    com.opt(opt($.inout_modifier), $._type, opt($.variadic_modifier), ','),
    ')',
    ':',
    field('return_type', $._type),
    ')',
  ),
```
These now work (though the second breaks syntax highlighting in VSCode):
```
(  ( /*test*/ function ( ) : void ) $test ) : void ==> {
    
};

(  ( function (  )  /*test*/  : void ) $test ) : void ==> {
    
};
```
But this still fails (and also breaks syntax highlighting ):
```
(  ( function /*test*/ (  ) : void ) $test ) : void ==> {
    
};
```

I think especially with that small of an issue area, I would be good with merging after the described update and documenting the issue. Thoughts?
Note: we can choose any two string tokens to break comment support between (aka `(function` or `function(` or `:(`).

It would also be fixed if anonymous_function_expression was ever deleted.

Reply:
Also I run against real-world code for each PR (albeit internal repos) to always verify the error count goes down. This fix did resolve all the related errors and didn't add any as far as I can tell. I can follow up more about this process in the issues you opened, but the short of it is that they run against the repos here: https://github.com/returntocorp/ocaml-tree-sitter-languages/blob/main/lang/hack/projects.txt

Reply:
The compromise seems good to merge. We can fix the last case separately.